### DISCUSSION TITLE
Enhance requireModule to set a variable indicating whether or not the module was installed

### DISCUSSION THREAD

**Discussion by rkeithhill** [PowerShellOrg/Plaster#390] [Ideas]
Saw this use case come up during @dchristian3188 user group presentation. I think it is a good idea.

**Comment by LaurentDardenne** [comment]
Is it not related to [this](https://github.com/PowerShell/Plaster/issues/234#issuecomment-278946504) ?

**Comment by rkeithhill** [comment]
Not quite.  The idea is to be able to use a Plaster variable created by the `<requireModule>` directive within another directive's condition (must appear after the requireModule) in the template.  Perhaps another conditional message that gets display if the required module was not found.  This may be a generalized concept where certain directives that could have different outcomes would get an attribute like `<requireModule name="posh-git" resultVariable="PoshGitStatus"/>` that you could use later with say `... condition="$PLASTER_RESULT_PoshGitStatus"`.  This is just early thinking and may change significantly (or not).   BTW I'm not 100% committed to this yet.  Toying with the idea atm.
